### ISSUE TITLE
`require/expose` does not work on macros

### ISSUE THREAD

**Issue by lihebi** [racket/rackunit#152]
[`require/expose`](https://docs.racket-lang.org/rackunit/api.html#%28form._%28%28lib._rackunit%2Fmain..rkt%29._require%2Fexpose%29%29) can require an unprovided binding from a module. But it only works on functions, not macros. E.g.

```rkt
(module aaa racket
  (define (foo a b) (+ a b))
  (define-syntax-rule (myadd x y)
    (+ x y)))
```

```rkt
(require rackunit)
;; OK
(require/expose 'aaa (foo))
;; ERROR: myadd: use does not match pattern: (myadd x y)
(require/expose 'aaa (myadd))
```

The relative code blocks implementing `require/expose`:

https://github.com/racket/rackunit/blob/6c04ee267d258a13f9c5cbc437dac60234e54ef4/rackunit-lib/rackunit/private/util.rkt#L72-L77

The reason is quite simple: transformer bindings are not available at runtime thus `(apply eval names)` thows error. Is it possible to make `require/expose` work on macros?

**Comment by jackfirth** [comment]
I don't think you need `require/expose` for that. You can use `dynamic-rerequire` instead, possibly combined with `all-defined-out`. You might also be interested in [`racket-reloadable`](https://github.com/tonyg/racket-reloadable-example), which uses `dynamic-rerequire` to implement hot code reloading for webserver-like programs.

**Comment by lihebi** [comment]
Thanks for the pointers! If I understand correctly, you mean:

```rkt
;; define module A with all-defined-out
(module A racket
  (provide (all-defined-out))
  (define a 1))
;; dynamically add a new binding b
(enter! 'A)
(define b 2)
(enter! #f)

;; try to reload
(dynamic-rerequire ''A)

;; OK
(dynamic-require ''A 'a)

;; ERROR: dynamic-require: name is not provided, name: 'b
(dynamic-require ''A 'b)

;; OK
(require/expose 'A (b))
```

Looks like this is not working, `all-defined-out` cannot pick up a new dynamically added binding. `b`.
